service/history/shard/context_impl.go

```diff
@@ -1203,18 +1207,23 @@ func (s *ContextImpl) updateShardInfo(
 		return err
 	}
 
+	s.tasksCompletedSinceLastUpdate += tasksCompleted
 	updateFnLocked()
 	s.shardInfo.StolenSinceRenew = 0
 
 	now := s.timeSource.Now()
-	if s.lastUpdated.Add(s.config.ShardUpdateMinInterval()).After(now) {
+	notFirstUpdate := !s.lastUpdated.IsZero()
```

Comment:
Trying to understand why firstUpdate or not matters here. 
Is it just for initializing `s.lastUpdated`? If so, can we do that when shard context object is created?

Reply:
There was probably a reason I added this but I deleted and all passing tests continued to pass, so I've deleted it